cmd/auth-handler.go

```diff
@@ -153,7 +154,7 @@ func checkRequestAuthType(ctx context.Context, r *http.Request, action policy.Ac
 	var locationConstraint string
 	if action == policy.CreateBucketAction {
 		// To extract region from XML in request body, get copy of request body.
-		payload, err := ioutil.ReadAll(r.Body)
+		payload, err := ioutil.ReadAll(io.LimitReader(r.Body, maxBucketPolicySize))
```

Comment:
maxBucketPolicySize should not be used as it is not related, you can define something else, maybe maxLocationConstraintSize

Reply:
Agree that this is cleaner. Will fix it. Thanks for the hint @krishnasrinivas